main/src/library/Condition.flix

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 Jonathan Lindegaard Starup
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+///
+/// A wrapper around a Java Condition.
+///
+opaque type Condition = ##java.util.concurrent.locks.Condition
+
+pub enum ConditionError {
+    case MonitorStateError(##java.lang.IllegalMonitorStateException)
+    case InterruptedError(##java.lang.InterruptedException)
+}
+
+namespace Condition {
+
+    ///
+    /// Causes the current thread to wait until it is signalled or interrupted.
+    ///
+    pub def await(condition: Condition): Result[Unit, ConditionError] & Impure =
+        import java.util.concurrent.locks.Condition.await();
+        let Condition(c) = condition;
+        try {
+            Ok(c.await())
+        } catch {
+            case e: ##java.lang.InterruptedException => Err(InterruptedError(e))
+            case e: ##java.lang.IllegalMonitorStateException => Err(MonitorStateError(e))
+        }
+
+    // TODO: Should this just be skipped?
+    // boolean await(long time, TimeUnit unit) throws InterruptedException
```

Comment:
Some of these methods seems to be more trouble to expose than it's worth, It's also easier to add later instead of breaking compatability

Reply:
I would focus on those that we need. We dont have to go for completeness.

Reply:
Okay, I've kept the ones that seem plausibly usable in relation to channels